src/Config/RabbitMQExtensionConfigProvider.cs

```diff
@@ -0,0 +1,110 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Description;
+using Microsoft.Azure.WebJobs.Host.Bindings;
+using Microsoft.Azure.WebJobs.Host.Config;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using RabbitMQ.Client;
+using RabbitMQ.Client.Events;
+
+namespace Microsoft.Azure.WebJobs.Extensions.RabbitMQ
+{
+    [Extension("RabbitMQ")]
+    internal class RabbitMQExtensionConfigProvider : IExtensionConfigProvider
+    {
+        private ILogger _logger;
+        private readonly IOptions<RabbitMQOptions> _options;
+        private readonly ILoggerFactory _loggerFactory;
+
+        public RabbitMQExtensionConfigProvider(IOptions<RabbitMQOptions> options, ILoggerFactory loggerFactory)
+        {
+            _options = options;
+            _loggerFactory = loggerFactory;
+        }
+
+        public void Initialize(ExtensionConfigContext context)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException("context");
+            }
+
+            _logger = _loggerFactory.CreateLogger("RabbitMQOutput");

```

Comment:
use `_loggerFactory.CreateLogger<RabbitMQExtensionConfigProvider>()`

Reply:
@brettsam - do you have any guidance on loggers for extensions? Should each class create a logger?